### PR TITLE
Azure OpenAI: openapi3 and tsp projectedName cleanup

### PR DESCRIPTION
**Pull request by trrwilson**
This change is intended to make Azure OpenAI's specification material more reviewable and accessible. It includes two notable arcs:

- New and first-used repository support for the openapi3 emitter
- Standardization of `@projectedName` decorator use within the Azure OpenAI TypeSpec specification

## OpenAPI3

OpenAI's source specification is written as an OpenAPI v3 YAML document: https://github.com/openai/openai-openapi/blob/master/openapi.yaml

In a shared effort to streamline the ingestion process and converge spec patterns, we'd also like to migrate to the more expressive OpenAPI3 document format.

**This change adds the `@typespec/openapi3` package to the repository**, as it was not previously included, and then adds emission of openapi3 YAML documents for Azure OpenAI in parallel to the existing openapi2 JSON documents. Future changes will further consolidate the source JSON/YAML documents and generated JSON/YAML documents, with the eventual goal being that only the YAML files generated from TypeSpec are necessary.

## ProjectedName use

Based on feedback, per-library-language customizations of field names have had their decorators moved from in-line with the service specification into the `client.tsp` file. Redundant/superfluous uses of `@projectedName` were also removed, leaving only wire-format casing differences (snake_case from OpenAI vs. typical camelCased for TypeSpec) as in-line applications of the decorator.

### CODE REVIEW

File: specification/cognitiveservices/OpenAI.Inference/client.tsp [hunk 1]
```diff
@@ -17,3 +17,189 @@ using Azure.ClientGenerator.Core;
 @@include(Azure.OpenAI.ImageGenerationOptions);
 @@include(Azure.OpenAI.ImageLocation);
 @@include(Azure.OpenAI.ImageGenerations);
+
```
Comment (jpalvarezl) [inline specification/cognitiveservices/OpenAI.Inference/client.tsp:20]: Nitpick: I can see how the `client.tsp` file could easily get out of hand should all the customization land in it. Would it be possible to split it? Could be a "functional" split where we have `renames.tsp` and `visibility.tsp` customizations. I am not aware of any guidelines regarding this, maybe we are free to do what feels cleanest to us.

File: specification/cognitiveservices/data-plane/AzureOpenAI/inference/2022-12-01/examples/completions.json
Comment (jpalvarezl) [inline specification/cognitiveservices/data-plane/AzureOpenAI/inference/2022-12-01/examples/completions.json:1]: It seems that the structure was flattened a bit. As I understand it, we did away with the `stable` and `preview` parent folders. For the `stable` ones then we won't have neither in the name nor the path an indication that is in fact a `stable` release. Are establishing `stable` as the default type of release? (therefore it doesn't need to be reflected in the path of the file)

File: specification/cognitiveservices/data-plane/AzureOpenAI/inference/2023-09-01-preview/generated.yaml [hunk 2]
```diff
@@ -0,0 +1,1914 @@
+openapi: 3.0.0
```
Comment (jpalvarezl) [inline specification/cognitiveservices/data-plane/AzureOpenAI/inference/2023-09-01-preview/generated.yaml:1]: I just wanted to express some happiness about the readability of yaml 😄 and the OpenAPI version bump too!

**Review by jpalvarezl — APPROVED** [review]
Really happy about this change! I am only left with a question: What happens to the `example.json` files with the new yaml files? I was not able to find how these are used in the new `generated.yaml` files. (Also not sure if they are meant to be used)

**Review by wangyuantao — APPROVED** [review]

File: specification/cognitiveservices/OpenAI.Inference/client.tsp (hunk 1 above)
Comment (trrwilson) [reply specification/cognitiveservices/OpenAI.Inference/client.tsp:20]: It's definitely possible and I agree with you -- this file's getting unruly! I'd like to discuss how we should break down client customizations across in a similar manner to operation/model decomposition (do we do it across operations/areas? across languages?) and I'd like to treat that as a *dedicated* discussion; I think having the change done (and functional) this way is a good stepping stone to having a productive talk on the topic.

File: specification/cognitiveservices/data-plane/AzureOpenAI/inference/2022-12-01/examples/completions.json
Comment (trrwilson) [reply specification/cognitiveservices/data-plane/AzureOpenAI/inference/2022-12-01/examples/completions.json:1]: Per the TypeSpec discussion channel, this flattening was reverted for now. That is the pattern we've had thus far, though: public API versions ending in `-preview` are preview, public API versions that are just the date label are stable/GA.

**Comment by trrwilson** [comment]
PR status note: this is just waiting on the fix for duplicate error definitions across sharedroutes (already confirmed via nightlies) to merge.

**Pull request closed without merging**